rio_tiler/mosaic/reader.py

```diff
@@ -83,15 +82,10 @@ def reader(asset: str, *args, **kwargs) -> Tuple[numpy.ndarray, numpy.ndarray]:
 
     for chunks in _chunks(assets, chunk_size):
         tasks = create_tasks(reader, chunks, threads, *args, **kwargs)
-        for (t, m), asset in filter_tasks(
-            tasks, allowed_exceptions=allowed_exceptions,
-        ):
+        for img, asset in filter_tasks(tasks, allowed_exceptions=allowed_exceptions,):
             assets_used.append(asset)
-            t = numpy.ma.array(t)
-            t.mask = m == 0
-
-            pixel_selection.feed(t)
+            pixel_selection.feed(img.as_masked())
             if pixel_selection.is_done:
-                return pixel_selection.data, assets_used
+                return ImageData(*pixel_selection.data, assets=assets_used), assets_used  # type: ignore
```

Comment:
`ImageData`. doesn't have the `crs` nor `bounds` for the `reader` result. I guess we could forward it from the last `img`